edgelet/iotedged/src/lib.rs

```diff
@@ -515,14 +517,27 @@ where
     let (mgmt_tx, mgmt_rx) = oneshot::channel();
     let (work_tx, work_rx) = oneshot::channel();
 
-    let mgmt = start_management(&settings, &runtime, &id_man, mgmt_rx);
+    let cert_manager = CertificateManager::new(crypto.clone()).context(ErrorKind::Initialize(
+        InitializeErrorReason::CreateTlsCertificate,
```

Comment:
We don't actually create a certificate till `get_certificate` is called no? If we change `CertificateManager::new` so that it no longer returns a `Result` then I suppose this error handling will go away.